joern-cli/frontends/rubysrc2cpg/src/test/scala/io/joern/rubysrc2cpg/querying/MethodTests.scala

```diff
@@ -256,4 +256,55 @@ class MethodTests extends RubyCode2CpgFixture {
 
   }
 
+  "Singleton Methods for module scope" should {
+    val cpg = code("""
+        |module F
+        | def F.bar(x)
+        |   x
+        | end
+        |end
+        |
+        |def F.baz(x)
+        |  x
+        |end
+        |
+        |F::bar(p)
+        |""".stripMargin)
+
+    "exist under the module TYPE_DECL" in {
+      inside(cpg.typeDecl.name("F").method.l) {
+        case bar :: baz :: Nil =>
+          inside(bar.parameter.l) {
+            case thisParam :: xParam :: Nil =>
+              thisParam.name shouldBe "F"
+              thisParam.typeFullName shouldBe "Test0.rb:<global>::program.F"
+
+              xParam.name shouldBe "x"
+            case xs => fail(s"Expected two parameters, got ${xs.name.mkString(", ")}")
+          }
+
+          inside(baz.parameter.l) {
+            case thisParam :: xParam :: Nil =>
+              thisParam.name shouldBe "F"
```

Comment:
```suggestion
              thisParam.name shouldBe "this"
```
Then the `.code` can be `F`